compiler/src/emit/native.rs

```diff
@@ -1,24 +1,32 @@
 use analyzer::relations::NativeId;
-use analyzer::types::hir::TypedExpr;
+use analyzer::types::hir::{MethodCall, TypeId};
 
 use crate::bytecode::{Instructions, Opcode};
 use crate::constant_pool::ConstantPool;
-use crate::emit::{emit, EmissionState, EmitterContext};
+use crate::emit::{emit, is_primitive, EmissionState, EmitterContext};
 use crate::locals::LocalsLayout;
 use crate::r#type::ValueStackSize;
 
 const STRING_EQ: &str = "lang::String::eq";
 const STRING_CONCAT: &str = "lang::String::concat";
 const INT_TO_STRING: &str = "lang::Int::to_string";
 const FLOAT_TO_STRING: &str = "lang::Float::to_string";
+const VEC_INDEX: &str = "lang::Vec::[]";
+const VEC_PUSH: &str = "lang::Vec::push";
+const VEC_LEN: &str = "lang::Vec::len";
+const STRING_SPLIT: &str = "lang::String::split";
+const STRING_BYTES: &str = "lang::String::bytes";
 
 /// Emits a primitive sequence of instructions.
-
 #[allow(clippy::too_many_arguments)]
 pub(crate) fn emit_natives(
     native: NativeId,
-    callee: &TypedExpr,
-    args: &[TypedExpr],
+    MethodCall {
+        callee,
+        arguments: args,
+        ..
+    }: &MethodCall,
```

Comment:
only `callee` is used
```suggestion
    callee: &TypedExpr,
```

Reply:
`args` is used every time a parameter need to be emitted.

Reply:
damn